analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,248 @@
+                                                       && programState.HasConstraint(symbolicValue, ConstantByteArraySymbolicValueConstraint.Constant),
+                    _ => false
+                };
+
+            private SymbolicValue GetSymbolicValue(InvocationExpressionSyntax invocation, ProgramState programState)
+            {
+                var invocationTarget = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
+                var targetSymbol = semanticModel.GetSymbolInfo(invocationTarget).Symbol;
+
+                return programState.GetSymbolValue(targetSymbol);
+            }
+
+            private bool IsIVMemberAccess(MemberAccessExpressionSyntax memberAccess) =>
+                memberAccess.NameIs("IV")
+                && semanticModel.GetSymbolInfo(memberAccess).Symbol.ContainingType.IsAny(VulnerableTypes);
+
+            private static bool HasInvalidConstraints(SymbolicValue value, ProgramState programState) =>
+                HasNotInitializedIVConstraint(value, programState) || HasNotInitializedKeyConstraint(value, programState);
+
+            private static bool HasNotInitializedIVConstraint(SymbolicValue value, ProgramState programState) =>
+                programState.Constraints[value].HasConstraint(IVInitializationSymbolicValueConstraint.NotInitialized);
+
+            private static bool HasNotInitializedKeyConstraint(SymbolicValue value, ProgramState programState) =>
+                programState.Constraints[value].HasConstraint(KeyInitializationSymbolicValueConstraint.NotInitialized);
+
+            private static bool IsCreateMethod(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+                invocation.IsMemberAccessOnKnownType("Create", VulnerableTypes, semanticModel);
+
+            private static bool IsCreateEncryptorMethod(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+                invocation.IsMemberAccessOnKnownType("CreateEncryptor", VulnerableTypes, semanticModel);
+
+            private static bool IsGenerateKeyMethod(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+                invocation.IsMemberAccessOnKnownType("GenerateKey", VulnerableTypes, semanticModel);
+
+            private static bool IsGenerateIVMethod(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+                invocation.IsMemberAccessOnKnownType("GenerateIV", VulnerableTypes, semanticModel);
+
+            private static bool IsRngCryptoServiceProviderSanitizer(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+                invocation.Expression is MemberAccessExpressionSyntax memberAccessExpressionSyntax
+                && (memberAccessExpressionSyntax.NameIs("GetBytes") || memberAccessExpressionSyntax.NameIs("GetNonZeroBytes"))
+                && semanticModel.GetSymbolInfo(invocation).Symbol.ContainingType.Is(KnownType.System_Security_Cryptography_RNGCryptoServiceProvider);
```

Comment:
NRE can probably happen (similar to `IsIVMemberAccess` comment), I didn't try to reproduce this one.
```suggestion
                && semanticModel.GetSymbolInfo(invocation).Symbol?.ContainingType.Is(KnownType.System_Security_Cryptography_RNGCryptoServiceProvider);
```